### ISSUE TITLE
Rancher-agent cannot provision custom node

### ISSUE THREAD

**Issue by imilchev** [rancher/rancher#34676]
**Rancher Server Setup**
- Rancher version: v2.6.0
- Installation option (Docker install/Helm Chart): Docker
- Proxy/Cert Details: We use CloudFlare and rancher is running with `--no-cacert` argument

**Information about the Cluster**
- Kubernetes version: v1.21.4-rancher1-1
- Cluster Type (Local/Downstream):
   - Custom cluster running on a VPS

**Describe the bug**
Trying to provision a node that has the Control plane + etcd role for the cluster. That is the first node that I am provisioning for that cluster. After I run the docker command on the node nothing happens. The logs of the rancher-agent show:

```
INFO: Arguments: --server https://rancher.podkrepi.bg --token REDACTED --etcd --controlplane
INFO: Environment: CATTLE_ADDRESS=164.138.217.42 CATTLE_INTERNAL_ADDRESS= CATTLE_NODE_NAME=podkrepibg01 CATTLE_ROLE=,etcd,controlplane CATTLE_SERVER=https://rancher.podkrepi.bg CATTLE_TOKEN=REDACTED
INFO: Using resolv.conf: nameserver 195.191.148.254 nameserver 195.191.149.254
INFO: https://rancher.podkrepi.bg/ping is accessible
INFO: rancher.podkrepi.bg resolves to 104.21.39.70 172.67.143.164
time="2021-09-08T10:37:49Z" level=info msg="Listening on /tmp/log.sock"
time="2021-09-08T10:37:49Z" level=info msg="Rancher agent version v2.6.0 is starting"
time="2021-09-08T10:37:49Z" level=info msg="Option controlPlane=true"
time="2021-09-08T10:37:49Z" level=info msg="Option worker=false"
time="2021-09-08T10:37:49Z" level=info msg="Option requestedHostname=podkrepibg01"
time="2021-09-08T10:37:49Z" level=info msg="Option dockerInfo={2PVD:YQWV:I5FA:HNI4:RF3F:SZAD:7Q6O:VPPR:YBV5:DVXB:634I:4HCH 1 1 0 0 1 overlay2 [[Backing Filesystem extfs] [Supports d_type true] [Native Overlay Diff true] [userxattr false]] [] {[local] [bridge host ipvlan macvlan null overlay] [] [awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog]} true false true true true true true true true true true true false 31 true 41 2021-09-08T13:37:49.624742396+03:00 json-file cgroupfs 1 0 5.4.0-84-generic Ubuntu 20.04.3 LTS 20.04 linux x86_64 https://index.docker.io/v1/ 0xc0010fd570 2 4116353024 [] /var/lib/docker    podkrepibg01 [] false 20.10.8   map[io.containerd.runc.v2:{runc [] <nil>} io.containerd.runtime.v1.linux:{runc [] <nil>} runc:{runc [] <nil>}] runc {  inactive false  [] 0 0 <nil> []} false  docker-init {e25210fe30a0a703442421b0f60afac609f950a3 e25210fe30a0a703442421b0f60afac609f950a3} {v1.0.1-0-g4144b63 v1.0.1-0-g4144b63} {de40ad0 de40ad0} [name=apparmor name=seccomp,profile=default]  [] [WARNING: No swap limit support]}"
time="2021-09-08T10:37:49Z" level=info msg="Option customConfig=map[address:164.138.217.42 internalAddress: label:map[] roles:[etcd controlplane] taints:[]]"
time="2021-09-08T10:37:49Z" level=info msg="Option etcd=true"
time="2021-09-08T10:37:49Z" level=info msg="Connecting to wss://rancher.podkrepi.bg/v3/connect with token starting with 9vlplwmvvsdcgddsj4chgt76g9k"
time="2021-09-08T10:37:49Z" level=info msg="Connecting to proxy" url="wss://rancher.podkrepi.bg/v3/connect"
time="2021-09-08T10:37:49Z" level=info msg="Waiting for node to register. Either cluster is not ready for registering, cluster is currently provisioning, or etcd, controlplane and worker node have to be registered"
time="2021-09-08T10:37:51Z" level=info msg="Waiting for node to register. Either cluster is not ready for registering, cluster is currently provisioning, or etcd, controlplane and worker node have to be registered"
time="2021-09-08T10:37:53Z" level=info msg="Waiting for node to register. Either cluster is not ready for registering, cluster is currently provisioning, or etcd, controlplane and worker node have to be registered"
time="2021-09-08T10:37:56Z" level=info msg="Waiting for node to register. Either cluster is not ready for registering, cluster is currently provisioning, or etcd, controlplane and worker node have to be registered"
time="2021-09-08T10:37:58Z" level=info msg="Waiting for node to register. Either cluster is not ready for registering, cluster is currently provisioning, or etcd, controlplane and worker node have to be registered"
```

**To Reproduce**
What I did is create a cluster with the specified Kubernetes version and Weave as a CNI plugin, then get the command for Control plane + etcd node and run it on my VPS.

**Result**
Cluster stuck in Provisioning state in Rancher.

**Expected Result**
To have the node visible there.

**Additional context**
If I try to delete the stuck cluster it is not possible as Rancher is not able to clean all the finalizers in the namespace created in the local cluster.

**Comment by p7kdev** [comment]
I have had similar problems recently. I think it was because I tried to register the master nodes first. It only continued after I had also registered a worker node. My guess, therefore, is that all roles must be in place before he actually finishes configuring a node.

Maybe someone from the Rancher/Suse team can confirm this?

**Comment by imilchev** [comment]
@p7kdev this was, indeed, the case. However, I saw no warning message about this when creating the cluster in the UI. Perhaps that's a nice thing to improve

**Comment by superseb** [comment]
These are the two locations that indicate what needs to be done, and the agent log also clearly describes the option. Which one of these were not showing in your setup?

<img width="1589" alt="Screenshot 2021-09-23 at 09 09 53" src="https://user-images.githubusercontent.com/2620179/134467418-c189cef5-c4ae-48ec-87fa-fd357d848c11.png">

<img width="413" alt="Screenshot 2021-09-23 at 09 10 16" src="https://user-images.githubusercontent.com/2620179/134467443-e036f4e7-519d-4bcb-a97b-8782475ecbd3.png">

**Comment by imilchev** [comment]
For me the yellow warning from the first screenshot was not showing. I did not really check the message in the clusters overview. Perhaps it's worth mentioning that I had a Rancher v2.5.8 installed, which I upgraded to v2.6.0. However, I think that issue was kind of minor as I obviously didn't provision all the nodes and was expecting for something to happen. The warning would, indeed, help but the bigger issue for me was the fact that I could not delete the cluster after that.

So to reproduce this create a cluster with custom nodes and add a node that does not have all roles (I did it with a control plane + etcd but no worker). Then try to delete that cluster from Rancher. It gets stuck and you cannot do anything with it unless you manually start removing finalizers from the Rancher local cluster on the CRDs
